artichoke-backend/src/extn/core/time/args.rs

```diff
@@ -87,78 +87,84 @@ impl TimeArgs {
     }
 }
 
-pub fn as_time_args(interp: &mut Artichoke, args: &[Value]) -> Result<TimeArgs, Error> {
-    // TimeArgs are in order of year, month, day, hour, minute, second, micros.
-    // This is unless there are 10 arguments provided (`Time#to_a` format), at
-    // which points it is second, minute, hour, day, month, year. The number of
-    // expected parameters doesn't give this hint though.
-
-    match args.len() {
-        0 | 9 | 11.. => {
-            let mut message = br#"wrong number of arguments (given "#.to_vec();
-            message.extend(args.len().to_string().bytes());
-            message.extend_from_slice(b", expected 1..8)");
-            Err(ArgumentError::from(message).into())
-        }
-        1..=8 => {
-            // For 0..=7 params, we need to validate to_int
-            let mut result = TimeArgs::default();
-            for (i, arg) in args.iter().enumerate() {
-                // The eighth parameter is never used, and thus no conversion
-                // is needed
-                if i == 7 {
-                    continue;
-                }
+impl TryConvertMut<&[Value], Args> for Artichoke {
+    type Error = Error;
+
+    fn try_convert_mut(&mut self, args: &[Value]) -> Result<Args, Self::Error> {
+        // Args are in order of year, month, day, hour, minute, second, micros.
+        // This is unless there are 10 arguments provided (`Time#to_a` format),
+        // at which points it is second, minute, hour, day, month, year. The
+        // number of expected parameters doesn't give this hint though.
+
+        match args.len() {
+            0 | 9 | 11.. => {
+                let mut message = br#"wrong number of arguments (given "#.to_vec();
+                message.extend(args.len().to_string().bytes());
+                message.extend_from_slice(b", expected 1..8)");
+                Err(ArgumentError::from(message).into())
+            }
+            1..=8 => {
+                // For 0..=7 params, we need to validate to_int
+                let mut result = Args::default();
+                for (i, arg) in args.iter().enumerate() {
```

Comment:
we can add a `&` pattern here to remove some dereference symbol noise from the code below:

```rust
for (i, &arg) in args.iter().enumerate() {
```